sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/BasicAuthJcsmpSessionService.java

```diff
@@ -39,34 +43,43 @@
  * <p>This class provides a way to connect to a Solace broker and receive messages from a queue. The
  * connection is established using basic authentication.
  */
-public class BasicAuthJcsmpSessionService extends SessionService {
-  private final String queueName;
-  private final String host;
-  private final String username;
-  private final String password;
-  private final String vpnName;
+@AutoValue
+public abstract class BasicAuthJcsmpSessionService extends SessionService {
+  /** The name of the queue to receive messages from. */
+  public abstract @Nullable String queueName();
+  /** The host name or IP address of the Solace broker. Format: Host[:Port] */
+  public abstract String host();
+  /** The username to use for authentication. */
+  public abstract String username();
+  /** The password to use for authentication. */
+  public abstract String password();
+  /** The name of the VPN to connect to. */
+  public abstract String vpnName();
+
+  public static Builder builder() {
+    return new AutoValue_BasicAuthJcsmpSessionService.Builder().vpnName(DEFAULT_VPN_NAME);
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder queueName(@Nullable String q);
+
```

Comment:
This is qute, but could you rename that parameter to queueName?

Reply:
I can :), in upcoming commit.